lib/git-process.ts

```diff
@@ -83,6 +119,40 @@ export class GitProcess {
   }
 
   /**
+   * Execute a command and interact with the process outputs directly.
+   *
+   * The returned promise will reject when the git executable fails to launch,
+   * in which case the thrown Error will have a string `code` property. See
+   * `errors.ts` for some of the known error codes.
+   */
+  public static spawn(args: string[], path: string, options?: IGitSpawnExecutionOptions): ChildProcess {
+    let customEnv = { }
+    if (options && options.env) {
+      customEnv = options.env
+    }
+
+    const { env, gitLocation } = setupEnvironment(customEnv)
+
+    const spawnArgs = {
+      env,
+      cwd: path
+    }
+
+    const spawnedProcess = spawn(gitLocation, args, spawnArgs)
+
+    if (options && options.stdin) {
+      // See https://github.com/nodejs/node/blob/7b5ffa46fe4d2868c1662694da06eb55ec744bde/test/parallel/test-stdin-pipe-large.js
+      spawnedProcess.stdin.end(options.stdin, options.stdinEncoding)
+    }
+
+    if (options && options.processCallback) {
+      options.processCallback(spawnedProcess)
```

Comment:
I initially wanted this to be similar to `GitProcess.exec` which lets you manipulate the child process before you then `await` the result. But given we don't have that constraint here we can probably :fire: this too.